### PR TITLE
[Merged by Bors] - chore: robustifying for debug.byAsSorry (part 10)

### PR DESCRIPTION
**Pull request by edegeltje**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Basic.lean [hunk 1]
```diff
@@ -1050,16 +1050,16 @@ instance commRing [CommRing R] : CommRing R[X] :=
 
 section NonzeroSemiring
 
-variable [Semiring R] [Nontrivial R]
+variable [Semiring R]
 
-instance nontrivial : Nontrivial R[X] := by
+instance nontrivial [Nontrivial R]: Nontrivial R[X] := by
```
Comment (grunweg) [inline Mathlib/Algebra/Polynomial/Basic.lean:1055]:
```suggestion
instance nontrivial [Nontrivial R] : Nontrivial R[X] := by

Comment (grunweg) [reply Mathlib/Algebra/Polynomial/Basic.lean:1055]: There is another instance below, please also fix this.

File: Mathlib/Order/Interval/Set/Pi.lean [hunk 2]
```diff
@@ -53,20 +53,20 @@ section Nonempty
 
 variable [Nonempty ι]
```
Comment (grunweg) [inline Mathlib/Order/Interval/Set/Pi.lean:54]: Can you remove this line now?

File: Mathlib/RingTheory/Localization/Basic.lean [hunk 3]
```diff
@@ -1090,7 +1090,9 @@ section
 variable [Algebra Rₘ Sₘ] [Algebra R Sₘ] [IsScalarTower R Rₘ Sₘ] [IsScalarTower R S Sₘ]
 variable (S Rₘ Sₘ)
 
-theorem IsLocalization.map_units_map_submonoid (y : M) : IsUnit (algebraMap R Sₘ y) := by
+theorem IsLocalization.map_units_map_submonoid
+    [i: IsLocalization (Algebra.algebraMapSubmonoid S M) Sₘ] (y : M) :
```
Comment (grunweg) [inline Mathlib/RingTheory/Localization/Basic.lean:1094]: Pre-existing, but: can you remove the name `i`, please? We prefer to not pass such typeclasses as named arguments.
```suggestion
    [IsLocalization (Algebra.algebraMapSubmonoid S M) Sₘ] (y : M) :

File: Mathlib/RingTheory/Valuation/Integers.lean [hunk 4]
```diff
@@ -70,17 +70,15 @@ theorem one_of_isUnit' {x : O} (hx : IsUnit x) (H : ∀ x, v (algebraMap O R x)
       (algebraMap O R).map_mul, v.map_mul]
     exact mul_le_mul_left' (H (u⁻¹ : Units O)) _
 
-variable (hv : Integers v O)
-
-theorem one_of_isUnit {x : O} (hx : IsUnit x) : v (algebraMap O R x) = 1 :=
+theorem one_of_isUnit (hv : Integers v O) {x : O} (hx : IsUnit x) : v (algebraMap O R x) = 1 :=
```
Comment (grunweg) [inline Mathlib/RingTheory/Valuation/Integers.lean:73]: To me (not an expert on this area), this looks like an interesting argument order. Would putting `hv` after `hx` be better?

File: tmp.txt [hunk 5]
```diff
@@ -0,0 +1,23 @@
+Some required builds logged failures:
```
Comment (grunweg) [inline tmp.txt:1]: Can you remove this file, please?

**Review by grunweg — COMMENTED** [review]
Some minor comments. Thanks for working on this!

File: Mathlib/RingTheory/Valuation/Integers.lean (hunk 4 above)
Comment (semorrison) [reply Mathlib/RingTheory/Valuation/Integers.lean:73]: I agree it's best to preserve argument order where it is easy to, during these refactors.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**